CahnHilliard.C

```diff
@@ -90,7 +90,11 @@ void CahnHilliard::printLog () const
 void CahnHilliard::setMode (SIM::SolutionMode mode)
 {
   m_mode = mode;
-  primsol.resize(mode < SIM::RHS_ONLY && gammaInv == 0.0 ? 0 : 2);
+  // LR cannot handle an additional empty entry
+  if (gammaInv == 0.0)
+    primsol.resize(mode < SIM::RHS_ONLY ? 0 : 2);
+  else
+    primsol.resize(mode < SIM::RHS_ONLY ? 0 : 1);
```

Comment:
Not sure about this change and why it is needed. The added comment is a bit wage.
However, I think the following is more correct:
```
  primsol.resize(gammaInv != 0.0 ? 2 : (mode < SIM::RHS_ONLY ? 0 : 1));
```
That is, when using penalty approach (gamma != 0) primsol is always resized to 2 regardless of solution mode, since the panalty terms should be computed from the previous phase-field value and not the current value.
When not using penalty approach, we need the (current) phase-field value when doing recovery/residual calculation only.

Reply:
yes, that was the intended change. it is needed, because, as stated in the pr description and discussed in your office, lr cannot handle an empty vector.